builtin/credential/aws/path_role.go

```diff
@@ -892,9 +892,11 @@ func (b *backend) pathRoleCreateUpdate(ctx context.Context, req *logical.Request
 	defaultLeaseTTL := b.System().DefaultLeaseTTL()
 	systemMaxTTL := b.System().MaxLeaseTTL()
 	if roleEntry.TokenTTL > defaultLeaseTTL {
+		roleEntry.TokenTTL = defaultLeaseTTL
```

Comment:
How come we clamp TTL to at most the default? I would only expect clamping to be needed for max. Is `.DefaultLeaseTTL()` just a poorly named function?

Reply:
The function is correctly named, but this is existing behavior within the plugin that was never actually implemented. We can either remove it and only cap on max, or implement the original intent. Thoughts?

Reply:
In that case, personally I would err towards removing the warning rather than adding capping - it's the more correct and safer change.

Reply:
I agree with Tom's recommendation

Reply:
@jasonodonnell what do you mean by never implemented? The value from `b.System().DefaultLeaseTTL()` is determined by this method:

https://github.com/hashicorp/vault/blob/2e0dcb7c9cc1f1e63c6bfd9852fb93956eb051cf/vault/dynamic_system_view.go#L124-L127

Reply:
Sync'ed up with Vinay on this and it turns out that TTL calculation caps the max TTL but doesn't care about the `ttl` (or `token_ttl`) value as long as that's under the max. I'm fine with removing this. At some point we were capping the default but then removed it when we changed things to use tokenutil.